source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -224,6 +227,9 @@ void AuthenticatorImpl::verifyKey() {
     // Remove JWT from headers.
     curr_token_->removeJwt(*headers_);
   }
+  if (set_payload_cb_ && provider.payload_in_metadata()) {
+    set_payload_cb_(provider.issuer(), jwt_->payload_str_base64url_);
```

Comment:
Left over from https://github.com/envoyproxy/envoy/pull/4698#discussion_r224839868

Parse JSON payload into a Struct and use that as value.

Reply:
Done